app/src/main/java/com/github/leuludyha/ibdb/presentation/screen/auth/signin/AuthenticationProvider.kt

```diff
@@ -104,3 +119,30 @@ fun AuthenticationProvider(
 
     }
 }
+
+/**
+ * This method will update the fields from authentication context that can be update from firebase.
+ */
+private fun loadAuthenticationContextFromFirebase(authContext: AuthenticationContext) {
+    authContext.principal.userId = Firebase.auth.currentUser?.uid!!
+    authContext.principal.username =
+        Firebase.auth.currentUser?.displayName ?: authContext.principal.username
+    authContext.principal.profilePictureUrl = Firebase.auth.currentUser?.photoUrl.toString()
+    authContext.principal.phoneNumber = Firebase.auth.currentUser?.phoneNumber
+}
+
+/**
+ * This is implemented following the recommendations from [here](https://stackoverflow.com/questions/56709604/check-for-internet-connectivity-on-api29).
+ *
+ * @return true if a network is available. Both wifi and mobile are acceptable.
+ */
+private fun checkNetworkAvailable(context: Context): Boolean {
+    val connectivityManager =
```

Comment:
Shouldn't this function be in a separate file like `NetworkManager` or something similar since it may be accessed from a lot of different places in our app?

Reply:
Done in next commit